django/dkg/cancer/json_api/endpoints.py

```diff
@@ -0,0 +1,72 @@
+import json # need double quotes in json POST bodies
+from django.http import JsonResponse
+from cancer.models import TrainingArticle, InferenceArticle
+
+
+def train(request):
+    """
+    Receives a POST-request with the following body: ?
+
+    Stores the data only.
+
+    :param request: HTTP-request carrying all RIS article information.
+    :param persistence: persistence object.
+    :return: possible HTTP-status codes: 200, 400.
+    """
+    request_body = json.loads(request.body)
+    print request_body
+    # store all parsed articles
+    article = TrainingArticle(
+        title='some article',
+        abstract='abstract of article',
+        authors='a\tb',
+        keywords='4,harn-prog\t1,harn-pall'
+    )
+    article.save()
+
+    return JsonResponse({'foo': 'bar'})
+
+
+def inference(request):
+    """
+    Receives a POST-request with the following body: ?
+    and runs multi-label inference on it.
+
+    :param request: HTTP-request carrying all RIS articles (without keywords)
+    :param persistence: persistence object.
+    :return: json response with all of the predicted labels from the uploaded RIS articles.
+
+    Response body sample: ?
+    """
+    request_body = json.loads(request.body)
+    print request_body
+    # store all parsed articles for inference
+    article = InferenceArticle(
```

Comment:
Ist definitiv komisch zwei tabellen mit identischer datenstruktur zu haben, wenn ich jetzt nochmal draufsehe macht das auch wenig sinn.

vorschlag zum schema:

```
articles tabelle
  * id (primaerschluessel)
  * part_of_training_corpus (boolean flag, damit koennen wir dann auch artikel fuer die inferenz in dieser tabelle speichern)
  * title
  * abstract
  * authors
  * (verbleibende features)

labels tabelle:
  * id (primaerschluessel)
  * article_id (fremdschluessel auf articles.id)
  * label (string, name des eigentlichen labels)
  * user (nutzer, der dieses label assigned hat)
```

damit haetten wir dann eine tabelle in der alle artikel stehen und eine sekundaertabelle mit 1:n von artikeln zu labels.